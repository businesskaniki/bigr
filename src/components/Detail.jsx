/* eslint-disable camelcase */
import React from 'react';
import { BsFillArrowLeftCircleFill } from 'react-icons/bs';
import { NavLink } from 'react-router-dom';
import { useSelector } from 'react-redux';

function Detail() {
  const makeups = useSelector((state) => state.makeupdetail);
  const {
    brand,
    name,
    price, image_link, website_link, rating, category,
    product_type, description, created_at, updated_at,
  } = makeups;
  return (
    <div className="detailcontainer">
      <h2>
        {' '}
        <span>
          <NavLink to="/">
            <BsFillArrowLeftCircleFill className="navigation" />
          </NavLink>
          {' '}
        </span>
        {brand}
      </h2>
      <div
        style={{
          // eslint-disable-next-line camelcase
          backgroundImage: `url(${image_link})`,
        }}
        className="detail"
      />
      <div>
        <p className="nm">{name}</p>
        { !price || <p className="name">{`price in usd: ${price}$`}</p>}
        { !website_link || <p className="name">{`for more information follow this link ${website_link}`}</p>}
        { !rating || <p className="name">{`rated at ${rating} stars`}</p>}
        { !category || <p className="name">{` put under the category  ${category}`}</p>}
        { !product_type || <p className="name">{`product type ${product_type}`}</p>}
        { !created_at || <p className="name">{`posted on this foroum at: ${created_at}`}</p>}
        { !updated_at || <p className="name">{`updated on:  ${updated_at}`}</p>}
        { !description || <p className="desc">{`${description}`}</p>}
      </div>
    </div>
  );
}

export default Detail;
