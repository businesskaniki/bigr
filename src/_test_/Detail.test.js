import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../redux/configstore';
import Detail from '../components/Detail';

describe('test detail component', () => {
  test('detail should be in the document', () => {
    render(
      <Provider store={store}>
        <Router><Detail /></Router>
      </Provider>,
    );
    const home = screen.getByRole('list');
    expect(home).toBeInTheDocument();
  });
  test('detail should match snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router><Detail /></Router>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
