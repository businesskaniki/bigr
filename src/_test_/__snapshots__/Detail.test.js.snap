// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test detail component detail should match snapshot 1`] = `
<div
  className="detailcontainer"
  role="list"
>
  <h2>
     
    <span>
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
      >
        <svg
          className="navigation"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 0a8 8 0 1 0 0 16A8 8 0 0 0 8 0zm3.5 7.5a.5.5 0 0 1 0 1H5.707l2.147 2.146a.5.5 0 0 1-.708.708l-3-3a.5.5 0 0 1 0-.708l3-3a.5.5 0 1 1 .708.708L5.707 7.5H11.5z"
          />
        </svg>
      </a>
       
    </span>
  </h2>
  <div
    className="detail"
    style={
      Object {
        "backgroundImage": "url(undefined)",
      }
    }
  />
  <div>
    <p
      className="nm"
    />
  </div>
</div>
`;
